Libraries/LibWeb/HTML/Navigable.cpp

```diff
@@ -1873,6 +1872,13 @@ void Navigable::navigate_to_a_fragment(URL::URL const& url, HistoryHandlingBehav
     // 7. Let entryToReplace be navigable's active session history entry if historyHandling is "replace", otherwise null.
     auto entry_to_replace = history_handling == HistoryHandlingBehavior::Replace ? active_session_history_entry() : nullptr;
 
+    // When replacing, preserve the navigation API ID and key to maintain entry identity
+    // This ensures the Navigation API can still find and reference this entry after replacement
+    if (entry_to_replace) {
+        history_entry->set_navigation_api_id(entry_to_replace->navigation_api_id());
```

Comment:
It would be great to have one method to override both of those things, cuz now you are duplicating a lot of code